File: solutions/java/matrix/1/src/main/java/Matrix.java
import java.util.Arrays;


class Matrix {

    int[][] matrix;
    int rowSize;

    Matrix(String matrixAsString) {
        var rows = matrixAsString.split("\n");
        rowSize = rows.length;
        var colSize = rows[0].split(" ").length;

        matrix = new int[rowSize][colSize];

        int i = 0;
        for (String r : rows) {
            matrix[i++] = Arrays
                    .stream(r.split(" "))
                    .mapToInt(Integer::parseInt).toArray();
        }
    }

    int[] getRow(int rowNumber) {
        return matrix[rowNumber - 1];
    }

    int[] getColumn(int columnNumber) {
        var arr = new int[rowSize];
        int i = 0;
        for (int[] a : matrix) {
            arr[i++] = a[columnNumber - 1];
        }

        return arr;
    }
}

File: solutions/kotlin/matrix/1/src/main/kotlin/Matrix.kt
class Matrix(private val matrixAsString: String) {

    val matrix: List<List<Int>> = matrixAsString
        .split("\n").map {
            it.split(" ").map { s: String -> s.toInt() }.toList()
        }.toList()

    fun column(colNr: Int): List<Int> {
        return matrix.map { it[colNr-1] }.toList()
    }

    fun row(rowNr: Int): List<Int> {
        return matrix[rowNr - 1]
    }
}
